okta/resource_user.go

```diff
@@ -239,6 +245,15 @@ func resourceUserCreate(d *schema.ResourceData, m interface{}) error {
 		}
 	}
 
+	// group assigning can only happen after the user is created as well
+	if len(d.Get("group_memberships").([]interface{})) > 0 {
```

Comment:
We have a conversion helper that takes an `interface{}` and returns a `[]string`. This makes the code prettier and technically less error prone if I am reading the code right but it does mean it will loop twice. So sort of turns into a horse a piece depending on how you look at it. I usually sacrifice a few nanoseconds for pretty code.